reference/toolstacks/php/drupal/redis-drupal-8.md

```diff
@@ -0,0 +1,163 @@
+Platform.sh's environment variables configuration, so using `settings.php` directly
+is the recommended approach.
+
+Place the following at the end of `settings.php`. Note the inline comments, as you may wish to customize
+it further.  Also review the `README.txt` file that comes with the redis module,
+as it has a great deal more information on possible configuration options. For instance,
+you may wish to not use Redis for the persistent lock if you have a custom module 
+that needs locks to persist for more than a few seconds.
+
+The example below is intended as a "most common case".
+
+```php
+// Set redis configuration.
+if (!empty($_ENV['PLATFORM_RELATIONSHIPS']) && extension_loaded('redis')) {
+  $relationships = json_decode(base64_decode($_ENV['PLATFORM_RELATIONSHIPS']), TRUE);
+
+  if (!empty($relationships['redis'][0])) {
+    $redis = $relationships['redis'][0];
+
+    $settings['cache']['default'] = 'cache.backend.redis';
+    $settings['redis.connection']['host'] = $redis['host'];
+    $settings['redis.connection']['port'] = $redis['port'];
+
+    // Enable the redis server overrides, e.g. cache tags checksum.
+    // See that file, you might want to copy and customize it, it is meant as an example
+    // and will change as more backends are offered by the redis module.
+    $settings['container_yamls'][] = 'modules/contrib/redis/example.services.yml';
+
+    // Manually add the classloader path, this is required for the container cache bin definition below
+    // and allows to use it without the redis module being enabled.
+    $class_loader->addPsr4('Drupal\\redis\\', 'modules/contrib/redis/src');
+
+    // Use redis for container cache.
+    $settings['bootstrap_container_definition'] = [
+      'parameters' => [],
+      'services' => [
+        'redis.factory' => [
+          'class' => 'Drupal\redis\ClientFactory',
+        ],
+        'cache.backend.redis' => [
+          'class' => 'Drupal\redis\Cache\CacheBackendFactory',
+          'arguments' => ['@redis.factory', '@cache_tags_provider.container'],
+        ],
+        'cache.container' => [
+          'class' => '\Drupal\redis\Cache\PhpRedis',
+          'factory' => ['@cache.backend.redis', 'get'],
+          'arguments' => ['container'],
+        ],
+        'cache_tags_provider.container' => [
+          'class' => 'Drupal\redis\Cache\RedisCacheTagsChecksum',
+          'arguments' => ['@redis.factory'],
+        ],
+      ],
+    ];
+
+    // Explicitly set the fast backend for bootstrap, default, discover and config,
+    // otherwise this gets lost when redis is enabled.
+    $settings['cache']['bins']['bootstrap'] = 'cache.backend.chainedfast';
+    $settings['cache']['bins']['discovery'] = 'cache.backend.chainedfast';
+    $settings['cache']['bins']['config'] = 'cache.backend.chainedfast';
+    // Default does not use the fast chained backend by default, but core only uses it for small caches
```

Comment:
I see that I failed to describe this here ;)

Simply put, this is a workaround for this core bug: https://www.drupal.org/node/2753989, which I finally managed to create. Maybe you simply want to write that.

The part about default goes a step further, it is an optimization that we use on our sites because we noticed that it only contains small amounts of data (= small caches) that are often requested.

Redis is the cache backend provider, it doesn't *use* the cache. Everyone else does. The reason it is commented out here and has some confusing descriptions is because most of this configuration is based on what we are using and I quickly extended it a bit with a few additional comments. Feel free to remove the part about default completely, it possibly doesn't belong into a minimal configuration guide.

I will instead open a core issue to move more of the bootstrap-ish cache entries into the bootstrap bin.